### PR TITLE
Feat/met 2740 test sandbox

### PR DESCRIPTION
**Pull request by acerdas84**
Filtering duplicate record ids

Also:

- Added flags to enable or disable queue listeners in application.yml (default value is enabled)

### CODE REVIEW

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetGeneratorServiceImpl.java [hunk 1]
```diff
@@ -42,8 +43,8 @@ public Dataset generate(String id, String name, Country country, Language langua
             .language(language)
             .content(record)
             .build())
-        .collect(Collectors.toList());
+        .collect(toSet());
 
-    return new Dataset(id, recordObjectList);
+    return new Dataset(id, recordObjectList, totalRecords - recordObjectList.size());
```
Comment (jochen-vermeulen) [inline src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetGeneratorServiceImpl.java:48]: I don't think this is the right way to count duplicates. I think we should be looking at the record IDs alone (and the equals method in Record look at other properties as well). Perhaps two records with the same ID but different languages exist, but they should still be counted as duplicates I think.

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java [hunk 2]
```diff
@@ -38,15 +41,27 @@ public String createDataset(String datasetName, Country country, Language langua
...
+    // if there are duplicate records in the original list
+    if (dataset.getRecords().size() < records.size()) {
+      // adjust records qty to be equal to non duplicate records
+      entity.setRecordsQuantity(dataset.getRecords().size());
+      try {
+        datasetRepository.save(entity);
+      } catch (Exception e) {
+        throw new ServiceException("Error updating the dataset: " + e.getMessage(), e);
+      }
+    }
```
Comment (jochen-vermeulen) [inline src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java:62]: Unless I'm mistaken, I think you can do this check before saving the entity the first time. I think that would be better... first create the entity completely with all counters and then save it once?

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetGeneratorServiceImpl.java (hunk 1 above)
Comment (acerdas84) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetGeneratorServiceImpl.java:48]: Yes I will update the equals method for the Record class

File: src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java (hunk 2 above)
Comment (acerdas84) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java:62]: In line 50 I need the datasetId to call generatorService, which is in charge of creating the Set<Records> (by reading the List< ByteArrayInputStream>). Each Record object contains the datasetId, then after calling generatorService I know how many records are duplicates, that is why I update the entity here, but I was not able to produce all counters at first.

Comment (jochen-vermeulen) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java:62]: You're right, I missed that. But still there must be a way... but perhaps that's not the most important thing to work on right now. This only occurs when there are duplicates, which should not happen too often.

Comment (acerdas84) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java:62]: Yes I thought it should not be often,  but I will create a story to not forget about this

Comment (acerdas84) [reply src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java:62]: [MET-2760](https://europeana.atlassian.net/browse/MET-2760)

**Pull request merged**